crux-llvm/src/Crux/LLVM/Log.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Crux.LLVM.Log
+  ( CruxLLVMLogMessage (..),
+    cruxLLVMLogMessageToSayWhat,
+  )
+where
+
+import Crux.Log (SayLevel (..), SayWhat (..), cruxLogTag)
+import Data.Text as Text (Text, pack, unwords)
+
+data CruxLLVMLogMessage
+  = ClangInvocation [Text]
+  | DebugBreakpoint Text
+  | DebugExecutable Text
+  | FailedToBuildCounterexampleExecutable
+  | SimulatingFunction Text
+  | UsingPointerWidthForFile Integer Text
+
+clangLogTag :: Text
+clangLogTag = "CLANG"
+
+cruxLLVMLogMessageToSayWhat :: CruxLLVMLogMessage -> IO SayWhat
```

Comment:
See note below: if we remove `IO` from these, they can just be a contravariant functor that the crux LLVM front-end contramaps onto the base logger.